api/v1alpha1/jobset_webhook.go

```diff
@@ -81,7 +83,23 @@ func (js *JobSet) ValidateCreate() error {
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (js *JobSet) ValidateUpdate(old runtime.Object) error {
-	return nil
+	oldObjCopy := old.DeepCopyObject().(*JobSet)
+	if pointer.BoolDeref(oldObjCopy.Spec.Suspend, false) {
+		return nil
+	}
+	var allErr []error
+	var updatableFields = []string{
+		"`template.spec.template.spec.nodeSelector`",
+	}
+	for index := range js.Spec.ReplicatedJobs {
+		oldObjCopy.Spec.ReplicatedJobs[index].Template.Spec.Template.Spec.NodeSelector = js.Spec.ReplicatedJobs[index].Template.Spec.Template.Spec.NodeSelector
+		if !apiequality.Semantic.DeepEqual(oldObjCopy.Spec.ReplicatedJobs, js.Spec.ReplicatedJobs) {
+			allErr = append(allErr, fmt.Errorf(
+				fmt.Sprintf("%s update validation: spec.ReplicatedJobs updates may not change fields other than %s", js.Spec.ReplicatedJobs[index].Name, strings.Join(updatableFields, ", ")),
+			))
+		}
+	}
+	return errors.Join(allErr...)
```

Comment:
```suggestion
	mungedSpec := js.Spec.DeepCopy()
	oldSpec := old.(*JobSet).Spec
	var updatableFields = []string{
		"`template.spec.template.spec.nodeSelector` when jobset is suspended",
	}
	if pointer.BoolDeref(oldObjCopy.Spec.Suspend, false) {
		for index := range js.Spec.ReplicatedJobs {
			mungedCopy.ReplicatedJobs[index].Template.Spec.Template.Spec.NodeSelector = oldSpec.ReplicatedJobs[index].Template.Spec.Template.Spec.NodeSelector
			}
		}
	}
	if !apiequality.Semantic.DeepEqual(oldSpec, mungedSpec) {
	  specDiff := cmp.Diff(oldSpec, mungedSpec)
		return field.Forbidden(specPath, fmt.Sprintf("jobset updates may not change fields other than %s\n%v", strings.Join(updatableFields, ","), specDiff))
	}
	return nil
```

Reply:
you need to import:
	- "github.com/google/go-cmp/cmp"
	- "k8s.io/apimachinery/pkg/util/validation/field"

Reply:
for `specPath`, just replace it with `field.NewPath("spec")`